src/System.Text.Json/tests/Serialization/TestClasses.Polymorphic.cs

```diff
@@ -129,6 +134,31 @@ public ObjectWithObjectProperties()
                 "Hello", "Again"
             };
 
+            IEnumerableT = new List<string>
+            {
+                "Hello", "World"
+            };
+
+            IListT = new List<string>
+            {
+                "Hello", "World"
+            };
+
+            ICollectionT = new List<string>
```

Comment:
What about other collections that are not list?

Reply:
I beleive we aren't supporting them now, but an extensibility model for enumerable collections is being designed https://github.com/dotnet/corefx/issues/36640.